### DISCUSSION TITLE
Output from inline PHP script

### DISCUSSION THREAD

**Discussion by franck-paul** [community/community#55163] [Actions] [answered]
### Select Topic Area

Question

### Body

Hi,

I have a workflow as follow wich use a PHP inline script to calculate and output two variables. These two variables will be used in the next steps:

```
name: Update dcstore.xml file if necessary

on:
  release:
    types:
      - released
  workflow_dispatch:

# A workflow run is made up of one or more jobs that can run sequentially or in parallel
jobs:
  # This workflow contains a single job called "dcstore"
  dcstore:
    # The type of runner that the job will run on
    runs-on: ubuntu-latest

    # Steps represent a sequence of tasks that will be executed as part of the job
    steps:
      # Setup PHP
      - name: Setup PHP
        uses: shivammathur/setup-php@v2
        with:
          php-version: '8.0'

      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
      - uses: actions/checkout@v2

      # Back to default branch
      - name: Back to default branch of repository
        shell: bash
        run: |
          echo "git fetch && git switch ${{github.event.repository.default_branch}}"
          git fetch
          git switch ${{github.event.repository.default_branch}}

      # Run PHP code
      - name: Run PHP code
        id: getversion
        shell: php {0}
        run: |
          <?php
          $version = '';
          $dcmin = '2.0';
          if (file_exists('dcstore.xml')) {
            $df = file_get_contents('./_define.php');
            if (preg_match('/registerModule\((.*?),(.*?)[\'\"],(.*?)[\'\"],(.*?)[\'\"](.*?)[\'\"](.*?)(,.*)\)/s',$df,$matches)) {
              if (isset($matches[5])) {
                $version = $matches[5];
                if (isset($matches[7])) {
                  $str = $matches[7];
                  if (preg_match('/\[(.*?)[\'\"]core[\'\"](.*?),(.*?)[\'\"](.*?)[\'\"](.*?)\]/s',$str,$submatches)) {
                    $dcmin = $submatches[4];
                  }
                }
              }
            }
            if ($version !== '') {
              $ds = file_get_contents('dcstore.xml');
              if ($ds) {
                $ds = preg_replace('/<version>(.*?)<\/version>/s',"<version>$version</version>",$ds);
                $ds = preg_replace('/download\/(.*?)\//s',"download/$version/",$ds);
                $ds = preg_replace('/(.*)-(.*?).zip/s',"$1-$version.zip",$ds);
                $ds = preg_replace('/<da:dcmin>(.*?)<\/da:dcmin>/s',"<da:dcmin>$dcmin</da:dcmin>",$ds);
                if ($ds) {
                  file_put_contents('dcstore.xml',$ds);
                }
              }
            }
          }
          echo "::set-output name=module_version::$version\n";
          echo "::set-output name=module_dcmin::$dcmin\n";

      # Cope with returned versions
      - name: Log output of the script
        shell: bash
        run: echo "${{steps.getversion.outputs.module_version}} (Dotclear ${{steps.getversion.outputs.module_dcmin}}+)"

     ...
```

I see today that the output should be change as the current way will be deprecated/removed soon. So I have to modify `echo "::set-output name=module_version::$version\n";` and `echo "::set-output name=module_dcmin::$dcmin\n";`.

I don't understand how to achieve this as the recommended way seems not functional in a PHP script (`echo "module_version=$version" >> $GITHUB_OUTPUT` and `echo "module_dcmin=$dcmin" >> $GITHUB_OUTPUT`) that seems legit as it's not really some PHP valid code.

How can I modify my script to output these two variables?

**Comment by airtower-luna** [comment]
`echo "module_version=$version" >> $GITHUB_OUTPUT` is shell syntax. `>>` redirects the output in *append* mode, `$GITHUB_OUTPUT` is the target file, with the twist that `$NAME` is variable access, so the actual file being redirected to is the one named in the variable.

What you need to do is *open the file named in the environment variable* `GITHUB_OUTPUT`, and *append* the `module_version=$version` line. I can't tell you how to do that in PHP, but maybe knowing the exact process you can figure it out. :slightly_smiling_face:

**Reply by franck-paul** [reply]
I tried `file_put_contents($_ENV["GITHUB_OUTPUT"], "module_version=$version\n", FILE_APPEND);` but it does not work as `$_ENV["GITHUB_OUTPUT"]` is not defined (null).

So if anybody knows how to get this variable in my script, it will be nice!

**Reply by airtower-luna** [reply, marked as answer]
From a quick search it looks like `_ENV` is sanitized by PHP in some way, try [`getenv()`](https://www.php.net/manual/en/function.getenv.php) instead.

**Reply by franck-paul** [reply]
Thank's a lot, you save my --life-- action \o/
